### PR TITLE
Fix/prevent twitter settings from sending empty screen name

### PR DESCRIPTION
**Pull request by santiagonoguez**
This makes the Twitter screen name required. The Save option won't be enabled unless an screen name is provided.

Staged version here, with twitter connection check disabled so the validation can be tested:

https://apps-stage-6.risevision.com/editor/workspace/be4fb65e-c2f3-43a7-b339-932222b49665/?cid=23aacc88-ff55-4af2-bb05-e104127b57b7

I also did a minor correction to README.md, so build instructions are properly formatted.

### CODE REVIEW

File: web/partials/widgets/twitter-settings-modal.html [hunk 1]
```diff
@@ -26,10 +26,10 @@ <h3 class="modal-title">{{"widget-twitter.settings" | translate }}</h3>
 	      <label class="control-label">{{"widget-twitter.screenNameText" | translate }}</label>
 		    <div class="input-group">
 		      <span class="input-group-addon" id="basic-addon1">@</span>
-		      <input id="twitterScreenName" type="text" class="form-control" placeholder="username" ng-model="settings.additionalParams.screenName">
+		      <input id="twitterScreenName" type="text" class="form-control" placeholder="username" required ng-model="settings.additionalParams.screenName">
```
Comment (alex-deaconu) [inline web/partials/widgets/twitter-settings-modal.html]: @santiagonoguez we usually add an asterisk (*) for required fields. Not sure if that applies for this. Thanks.

Comment (alex-deaconu) [reply web/partials/widgets/twitter-settings-modal.html]: Also, are we planning to show a validation error for failing to enter the required value?

Comment (santiagonoguez) [reply web/partials/widgets/twitter-settings-modal.html]: @alex-deaconu Thanks ! I'll add the validation error, to make it consistent with the required field of Calendar widget. I'll also add the asterisk.

File: web/partials/widgets/twitter-settings-modal.html [hunk 2]
```diff
@@ -46,13 +49,13 @@ <h4>{{"widget-twitter.previewLable" | translate }}</h4>
...
 
-		  
-		  
+
+
 
 		</form>
 	</div>
 	<div class="modal-footer">
-	  <widget-button-toolbar save="saveSettings()" cancel="closeSettings()" disable-save="settingsForm.$invalid || !twitterConnected">
```
Comment (jungeunyoon) [inline web/partials/widgets/twitter-settings-modal.html]: @santiagonoguez Don't forget to remove before merging, please!

Comment (santiagonoguez) [reply web/partials/widgets/twitter-settings-modal.html]: won't forget :)

**Comment by santiagonoguez** [comment]
@alex-deaconu I was experimenting by adding it, but actually liked better how it shows with the default error styling being applied to the regular label.

screenshot

Thoughts ? @sheadarlison @jungeunyoon 

Current behavior can be seen by selecting Twitter component in staged presentation:

https://apps-stage-6.risevision.com/editor/workspace/be4fb65e-c2f3-43a7-b339-932222b49665/?cid=23aacc88-ff55-4af2-bb05-e104127b57b7

**Comment by alex-deaconu** [comment]
@santiagonoguez okay, if we leave it like that, that means the i18n String needs to be removed from the CH PR as it is unused. Thanks.

**Comment by jungeunyoon** [comment]
@mmeiers Thoughts?

**Comment by santiagonoguez** [comment]
@alex-deaconu agreed, I'll remove it there if this PR if we move forward without that message.

**Comment by ghislaineguerin** [comment]
@santiagonoguez Is this warning enabled by default? It doesn't specify how to solve it and it relies mostly on color to indicate there's an issue. We need to be clear that the user has produced an error, as in "Error: you must X before you can Y'
@alex-deaconu FYI

**Comment by mmeiers** [comment]
@santiagonoguez @jungeunyoon 
I think disabling the save button is sufficient and consistent with other widgets (see webpage and spreadsheet widgets). I wouldnt apply an error message that shows right when a user opens the widgets settings. I had a discussion with Apps/ UX Designers previously and the conclusion was we shouldn't do this.

**Comment by santiagonoguez** [comment]
@ghislaineguerin OK, thanks ! 

Then I should specify the error below the input as:

  Error: you should specify a handler before you can save this configuration

Does that sound good ?
@sheadarlison @jungeunyoon @mmeiers

**Comment by santiagonoguez** [comment]
@alex-deaconu pristine is now applied, works like a charm ! 

Finally, I'll just update the validation message back in common-header to follow what @ghislaineguerin has suggested.

**Comment by santiagonoguez** [comment]
@alex-deaconu everything's up to date now !

Please review again.

**Comment by santiagonoguez** [comment]
@alex-deaconu Merging of this PR is blocked until you review / approve, thanks !

**Comment by alex-deaconu** [comment]
@santiagonoguez looks good, thanks for fixing!

**Pull request merged**